sqlglot/generator.py

```diff
@@ -564,21 +571,20 @@ def create_sql(self, expression: exp.Create) -> str:
                     if index.args.get("columns")
                     else ""
                 )
-                if index.args.get("primary") and properties_locs.get(
-                    exp.Properties.Location.POST_INDEX
-                ):
-                    postindex_props_sql = self.properties(
-                        exp.Properties(
-                            expressions=properties_locs[exp.Properties.Location.POST_INDEX]
-                        ),
-                        wrapped=False,
+                ind_sql = f"{ind_unique}{ind_primary}{ind_amp} INDEX{ind_name}{ind_columns}"
+
+                if indexes_sql:
+                    indexes_sql.append(ind_sql)
+                else:
+                    indexes_sql.append(
+                        f"{ind_sql}{postindex_props_sql}"
+                        if index.args.get("primary")
+                        else f"{postindex_props_sql}{ind_sql}"
                     )
```

Comment:
An idea to further simplify this:
```python
ind_sql = f"{ind_unique}{ind_primary}{ind_amp} INDEX{ind_name}{ind_columns}"
ind_sql = f"{ind_sql}{postindex_props_sql}" if index.args.get("primary") else f"{postindex_props_sql}{ind_sql}"
indexes_sql.append(ind_sql)
```

Reply:
Scratch that, I misread a part of the code. This won't work.

Reply:
Wouldn't this add in postindex_props_sql every time through the loop instead of just the first?